scene/resources/resource_format_text.cpp

```diff
@@ -2270,11 +2273,32 @@ Error ResourceFormatSaverTextInstance::save(const String &p_path, const Ref<Reso
 
 			f->store_line("]");
 
+			HashSet<String> stored_metadatas;
+			if (i == 0) {
+				// Store scene metadata as the first node's metadata.
+				List<PropertyInfo> property_list;
+				packed_scene->get_property_list(&property_list);
+				for (List<PropertyInfo>::Element *PE = property_list.front(); PE; PE = PE->next()) {
+					String name = PE->get().name;
+					if (name.begins_with("metadata/") && PE->get().usage & PROPERTY_USAGE_STORAGE) {
```

Comment:
```suggestion
					if (PE->get().usage & PROPERTY_USAGE_STORAGE && meta_name.begins_with("metadata/")) {
```
Cheaper condition first